randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -77,27 +81,28 @@ def _create_room_dict(self):
             for region in self.game.region_list.regions for area in region.areas
         }
 
-    def _create_starting_items_dict(self):
+    def _create_starting_items_dict(self) -> dict:
         starting_resources = self.patches.starting_resources()
         return {
             resource.long_name: quantity
             for resource, quantity in starting_resources.as_resource_gain()
         }
 
-    def _create_starting_location(self):
+    def _create_starting_location(self) -> dict:
         return {
             "save_room": self.game.region_list.node_by_identifier(self.patches.starting_location).extra["save_room"]
         }
 
-    def _create_hash_dict(self):
+    def _create_hash_dict(self) -> dict:
         return {
             "word_hash": self.description.shareable_word_hash,
             "hash": self.description.shareable_hash,
             "session_uuid": str(self.players_config.get_own_uuid())
         }
 
-    def _create_game_patches(self, pickup_list: list[ExportedPickupDetails], item_info: dict, rng: Random):
-        def get_locked_ammo_text(ammo_item: str):
+    def _create_game_patches(self, config: AM2RConfiguration, pickup_list: list[ExportedPickupDetails],
```

Comment:
Was there s need to pass it now as a parameter instead of accessing it via `self.configuration` ( or assign it in the method itself to a new variable `config` to use a shorter name)?

Reply:
I'm *slowly* trying to refactor the patch data factory a bit, to have the methods not as coupled to the factory. While poking around in dread/SM, and needing to write the test for music shuffling, I noticed that it's something that's helpful.